Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -108,16 +98,58 @@ - (NSURLSessionTaskState)state
 
 #pragma mark Helpers
 
+- (void)makeRequest
+{
+    CDTHTTPInterceptorContext *ctx =
+        [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    // We make sure all objects support `interceptRequestInContext:` during init.
+    for (NSObject<CDTHTTPInterceptor> *obj in self.requestInterceptors) {
+        ctx = [obj interceptRequestInContext:ctx];
+    }
+
+    __weak CDTURLSessionTask *weakSelf = self;
+    self.inProgressTask = [self.session
+        dataTaskWithRequest:ctx.request
+          completionHandler:^void(NSData *data, NSURLResponse *response, NSError *error) {
+            __strong CDTURLSessionTask *strongSelf = weakSelf;
+            if (strongSelf) {
+                // maybe we can reuse the context from earlier?
```

Comment:
i'm pretty sure you can re-use the last context after it was processed by the request interceptor